src/components/cardModal/CardModal.js

```diff
@@ -0,0 +1,141 @@
+import styles from "@/styles/cardModal.module.scss";
+import { useState, useContext } from "react";
+import { LoggedInContext } from "@/pages/_app";
+
+
+async function sendCardInfo(userName, cardInfo)
+{
+    const resp = await fetch('/api/users/addcard', {
+        "method": "PATCH",
+        "headers": {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        "body": JSON.stringify({
+            "userName": userName,
+            "paymentMethods": cardInfo
+        })
+    })
+    const data = await resp.json();
+
+    return data;
+}
+
+
+
+export default function CardModal2({ showCardModal, handleCloseCardModal, handleCard })
+{
+    const [cardInfo, setCardInfo] = useState({ "cardNr": '', "validTo": '', "CVC": '', "nameOnCard": '' })
+    const {
+        isLoggedIn,
+        setIsLoggedIn,
+        username,
+        setUsername,
+        setDisplayPopup,
+        setPopupTitle,
+        setPopupMessage
+    } = useContext(LoggedInContext);
+
+    if (!showCardModal)
+        return null;
+
+    return (
+        <>
+            <div className={styles.cardModalOverlay} onClick={handleCloseCardModal} />
+
+            <div className={styles.cardModalContainer}>
+                <button
+                    className={styles.closeBtn}
+                    onClick={handleCloseCardModal}
+                    aria-label="Stäng modalen knapp"
+                >X</button>
+
+                <form className={styles.cardModalForm} onSubmit={(ev) =>
+                {
+                    ev.preventDefault();
+                    handleCard(cardInfo);
+
+                    if (isLoggedIn)
+                    {
+                        const resp = sendCardInfo(username, cardInfo);
+                        if (resp)
+                        {
+                            setPopupTitle("Kort");
+                            setPopupMessage("Ditt kort är tillagt!");
+                            setDisplayPopup(true);
+                        } else
+                        {
+                            setPopupTitle("Kort");
+                            setPopupMessage("Det blev fel, försök igen!");
+                            setDisplayPopup(true)
+                        }
+                    }
+
+
+                    handleCloseCardModal();
+                }}>
+                    <div className={`${styles.cardModalSections} ${styles.cardModalSectionCard}`}>
+                        <label htmlFor="cardNumber">Kortnummer</label>
+                        <input
+                            id="cardNumber"
+                            type="tel"
+                            name="cardNumber"
+                            value={cardInfo.cardNr}
+                            onChange={(e) => setCardInfo({ ...cardInfo, "cardNr": e.target.value })}
+                            pattern='[\d| ]{16,22}'
+                            maxLength='19' placeholder="💳 0000 0000 0000 0000"
```

Comment:
I like the solution of using emojis for this, but I would put them in a separate element so that they don't get replaced by the content that the user puts into the input field.